action.nu

```diff
@@ -79,12 +79,25 @@ export module plugin-list {
     # converts repository url into raw download link for Cargo.toml
     def "get-raw-toml-address" [
         url: string, # github repository url (e.g. https://github.com/FMotalleb/nu_plugin_port_scan)
-        branch: string # branch name (e.g. main)
+        branch: string, # branch name (e.g. main)
+        software?: string,
     ]: nothing -> record {
         let url: record = ($url | url parse)
+
+        # Unspecified forge software is determined from the host.
+        let software = if $software == null {
+            let foo = match $url.host {
```

Comment:
Please update `foo` to a more descriptive variable name.

Reply:
Now that I think about it, couldn't this whole "software" section could just be a match statement. Maybe something like this?
```nushell
        # Unspecified forge software is determined from the host.
        let host = match $url.host {
                'github.com' => 'github',
                'gitlab.com' => 'gitlab',
                'codeberg.org' => 'forgejo',
                _ => $software
        }
```